### PR TITLE
Add a parameter in init which allows setting of Authorization header in the API calls

### PR DESCRIPTION
**Pull request by nogoodusername**

### CODE REVIEW

**Comment by dabeeeenster** [comment]
🤷

**Comment by dabeeeenster** [comment]
🤷

**Comment by kyle-ssg** [comment]
I like the idea however the PR is too specific I think, what would be better is if you could specify headers in init like:

```
flagsmith.init({
  headers : {"AUTHORIZATION":"x", "OtherHeader": "y"}
})

``

File: flagsmith/package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "flagsmith",
+  "name": "flagsmith-wi-fork",
   "version": "1.6.6",
   "description": "Feature flagging to support continuous development",
   "main": "./index.js",
   "repository": {
     "type": "git",
-    "url": "https://github.com/Flagsmith/flagsmith-js-client"
+    "url": "https://github.com/workindia/flagsmith-js-client"
```
Comment (kyle-ssg) [inline flagsmith/package.json:8]: 🐛

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "flagsmith",
+  "name": "flagsmith-wi-fork",
```
Comment (kyle-ssg) [inline package.json:2]: 🐛

**Comment by nogoodusername** [comment]
Hey,

I'll come back in few days with a generic solution

File: react-native-flagsmith/package.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "react-native-flagsmith",
+  "name": "react-native-flagsmith-wi-fork",
```
Comment (kyle-ssg) [inline react-native-flagsmith/package.json:2]: 🐛

**Comment by kyle-ssg** [comment]
@nogoodusername Awesome, thanks so much for spending the time doing this!

**Comment by nogoodusername** [comment]
This is the idea:

"Add an authorization flag to API calls made to self-hosted flagsmith server.
We're planning to run flagsmith behind a gateway and the gateway will ensure that no unauthorised identity / trait manipulation will happen via clients.

This PR accepts authToken in init and passes the token value in API calls via getJSON"

**Comment by kyle-ssg** [comment]
Yeah understood, It's just that auth headers take many forms, and this would be useful for others  (e.g. cache headers for custom cdn solutions)

**Comment by nogoodusername** [comment]
Agreed, will share a PR later.

I was not supposed to raise this PR on this repo, it was meant for our `workindia` organisation. I'll raise a generic implementation over the weekend. ✌️

**Pull request closed without merging**